stimela/kitchen/step.py

```diff
@@ -437,7 +434,7 @@ def run(
                 log_rich_payload(self.log, "current backend settings are", opts_yaml, syntax="yaml")
             backend_opts = OmegaConf.merge(StimelaBackendSchema, backend_opts)
             backend_opts = OmegaConf.to_object(backend_opts)
-            backend_runner = runner.validate_backend_settings(backend_opts, log=self.log)
+            backend_runner = runner.validate_backend_settings(backend_opts, self.cargo, log=self.log)
```

Comment:
@SpheMakh have you tested this with nested recipes? I have a sneaky suspicion this might fail if the cargo is a recipe, and the backend is singularity only.

I think the correct thing to do to pass ``self.cargo if isinstance(self.cargo, Cab) else None``.

I think that should be done when invoking in finalize() and run(). No need in build() since there the if statement checks for a Cab anyway.